### ISSUE TITLE
[BUG] v1.4.0-rc1 Doesn't Recogonize All Available Drives At Install

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#6694]
### Describe The Bug:
Something is happening where v1.4.0-rc1 is not recogonizing all available drives at install time.
It's only showing the first drive and will not display the rest within the interactive ISO installation.
This prevents users from selecting other possible disks.

### To Reproduce:
Pre-Reqs:
- have some sort of raid array controller I'm using the [HPE Smart Array P440/4GB Controller](https://www.hpe.com/psnow/doc/c04346277) with two logical drives set up, that are using two separate but same size disks, on an HP DL160 Gen 9
1. let interactive iso installer boot
1. try to select your other disk
1. you'll only see one option

### Expected Behavior:
This to behave the same as Harvester v1.2.1 (see attached video) where multiple logical drives on a raid controller were available at install.

### Environment:
 - Harvester ISO version: v1.4.0-rc1 
 - Underlying Infrastructure: HP DL160 Gen 9, P440 Smart Array Controller 4GB, two logical drives / two same size Kingston physical SSDs

### Additional Context:

**_P440 Setup on HP DL-160 Gen 9:_**

https://github.com/user-attachments/assets/61fd223a-29d9-496a-9bbe-914824b9f504

**_Harvester v1.2.1 Has No Issue Seeing The Multiple Logical Drives at Install Panel:_**

https://github.com/user-attachments/assets/bead5d33-dc65-4efe-bc4c-cfac2e97bad8

**_Harvester v1.4.0-rc1 has an issue seeing the multiple logical drives at install, it only sees one and only one:_**

https://github.com/user-attachments/assets/7db6e0b2-a9cc-436b-af56-0685f97bd5c5

**Comment by irishgordo** [comment]
Additionally this seems to pick a drive at random - so at times it might be the first logical drive as sda - at other times it might be the second logical drive as sda

**Comment by tserong** [comment]
@ibrokethecloud do you mind checking this? My guess from the description is it's maybe related to the disk deduplication introduced in https://github.com/harvester/harvester-installer/pull/778

**Comment by TachunLin** [comment]
I also tried the ISO installation v1.4.0-rc1 using `USB` install on my local baremetal machine. 
The Machine model is Dell Precision T5820 tower machine server. 


* There it could display most part of the disks, beside the strange `sdd` with `0B` 
  ![IMG_8055](https://github.com/user-attachments/assets/adc1f61e-d078-4164-8a1c-920e0d4c8c16)
  ![IMG_8056](https://github.com/user-attachments/assets/8fd00787-47d4-4ec9-8356-c8b8f51d0795)

I am thinking perhaps the tower machine not have RAID setup on these disks.

**Comment by irishgordo** [comment]
v1.3.2 also does not experience this, as an added mention

**Comment by Vicente-Cheng** [comment]
Hi @irishgordo

I am curious about the information on the raid device.
Could you help attach the udev info for those raid disks?
e.g. `udevadm info /dev/sda`

Thanks!

**Comment by irishgordo** [comment]
@Vicente-Cheng  - yeah let me get those details for you, give me just a little bit :smile: :+1:

**Comment by irishgordo** [comment]
@Vicente-Cheng 
Heres's a bit more detail.

## Loadout
- 2 Node HP DL160 Gen9s, I'll just gather node6 information as both nodes are the exact same 

### Node 6

**/dev/sda** :
```bash
P: /devices/pci0000:00/0000:00:01.0/0000:0d:00.0/host0/target0:1:0/0:1:0:1/block/sda
N: sda
L: 0
S: disk/by-id/scsi-SHP_LOGICAL_VOLUME_PDNMF0ARH1619G
S: disk/by-id/wwn-0x600508b1001c89f560d91370f6f778b2
S: disk/by-path/pci-0000:0d:00.0-scsi-0:1:0:1
S: disk/by-id/scsi-3600508b1001c89f560d91370f6f778b2
S: disk/by-id/scsi-0HP_LOGICAL_VOLUME_01000000
E: DEVPATH=/devices/pci0000:00/0000:00:01.0/0000:0d:00.0/host0/target0:1:0/0:1:0:1/block/sda
E: DEVNAME=/dev/sda
E: DEVTYPE=disk
E: DISKSEQ=1
E: MAJOR=8
E: MINOR=0
E: SUBSYSTEM=block
E: USEC_INITIALIZED=13841324
E: DONT_DEL_PART_NODES=1
E: ID_PATH=pci-0000:0d:00.0-scsi-0:1:0:1
E: ID_PATH_TAG=pci-0000_0d_00_0-scsi-0_1_0_1
E: SCSI_TPGS=0
E: SCSI_TYPE=disk
E: SCSI_VENDOR=HP
E: SCSI_VENDOR_ENC=HP\x20\x20\x20\x20\x20\x20
E: SCSI_MODEL=LOGICAL_VOLUME
E: SCSI_MODEL_ENC=LOGICAL\x20VOLUME\x20\x20
E: SCSI_REVISION=7.00
E: ID_SCSI=1
E: ID_SCSI_INQUIRY=1
E: SCSI_IDENT_SERIAL=PDNMF0ARH1619G
E: SCSI_IDENT_LUN_NAA_REGEXT=600508b1001c89f560d91370f6f778b2
E: SCSI_IDENT_LUN_VENDOR=01000000
E: ID_VENDOR=HP
E: ID_VENDOR_ENC=HP\x20\x20\x20\x20\x20\x20
E: ID_MODEL=LOGICAL_VOLUME
E: ID_MODEL_ENC=LOGICAL\x20VOLUME\x20\x20
E: ID_REVISION=7.00
E: ID_TYPE=disk
E: ID_WWN_WITH_EXTENSION=0x600508b1001c89f560d91370f6f778b2
E: ID_WWN=0x600508b1001c89f5
E: ID_BUS=scsi
E: ID_SERIAL=3600508b1001c89f560d91370f6f778b2
E: ID_SERIAL_SHORT=600508b1001c89f560d91370f6f778b2
E: ID_SCSI_SERIAL=PDNMF0ARH1619G
E: multipath=off
E: ID_PART_TABLE_UUID=d2a27388-955b-4279-94a1-5a5cb4b256e6
E: ID_PART_TABLE_TYPE=gpt
E: COMPAT_SYMLINK_GENERATION=2
E: DEVLINKS=/dev/disk/by-id/scsi-SHP_LOGICAL_VOLUME_PDNMF0ARH1619G /dev/disk/by-id/wwn-0x600508b1001c89f560d91370f6f778b2 /dev/disk/by-path/pci-0000:0d:00.0-scsi-0:1:0:1 /dev/disk/by-id/scsi-3600508b1001c89f560d91370f6f778b2 /dev/disk/by-id/scsi-0HP_LOGICAL_VOLUME_01000000
E: TAGS=:systemd:
E: CURRENT_TAGS=:systemd:
```

**/dev/sdb**:
```bash
P: /devices/pci0000:00/0000:00:01.0/0000:0d:00.0/host0/target0:1:0/0:1:0:0/block/sdb
N: sdb
L: 0
S: disk/by-id/scsi-SHP_LOGICAL_VOLUME_PDNMF0ARH1619G
S: disk/by-path/pci-0000:0d:00.0-scsi-0:1:0:0
S: disk/by-id/wwn-0x600508b1001c85cc8c95588d66ea63f3
S: disk/by-id/scsi-3600508b1001c85cc8c95588d66ea63f3
S: disk/by-uuid/8efc2e53-1d6e-4312-a19d-45b89faa6058
S: disk/by-id/scsi-0HP_LOGICAL_VOLUME_00000000
E: DEVPATH=/devices/pci0000:00/0000:00:01.0/0000:0d:00.0/host0/target0:1:0/0:1:0:0/block/sdb
E: DEVNAME=/dev/sdb
E: DEVTYPE=disk
E: DISKSEQ=2
E: MAJOR=8
E: MINOR=16
E: SUBSYSTEM=block
E: USEC_INITIALIZED=13837616
E: DONT_DEL_PART_NODES=1
E: ID_PATH=pci-0000:0d:00.0-scsi-0:1:0:0
E: ID_PATH_TAG=pci-0000_0d_00_0-scsi-0_1_0_0
E: SCSI_TPGS=0
E: SCSI_TYPE=disk
E: SCSI_VENDOR=HP
E: SCSI_VENDOR_ENC=HP\x20\x20\x20\x20\x20\x20
E: SCSI_MODEL=LOGICAL_VOLUME
E: SCSI_MODEL_ENC=LOGICAL\x20VOLUME\x20\x20
E: SCSI_REVISION=7.00
E: ID_SCSI=1
E: ID_SCSI_INQUIRY=1
E: SCSI_IDENT_SERIAL=PDNMF0ARH1619G
E: SCSI_IDENT_LUN_NAA_REGEXT=600508b1001c85cc8c95588d66ea63f3
E: SCSI_IDENT_LUN_VENDOR=00000000
E: ID_VENDOR=HP
E: ID_VENDOR_ENC=HP\x20\x20\x20\x20\x20\x20
E: ID_MODEL=LOGICAL_VOLUME
E: ID_MODEL_ENC=LOGICAL\x20VOLUME\x20\x20
E: ID_REVISION=7.00
E: ID_TYPE=disk
E: ID_WWN_WITH_EXTENSION=0x600508b1001c85cc8c95588d66ea63f3
E: ID_WWN=0x600508b1001c85cc
E: ID_BUS=scsi
E: ID_SERIAL=3600508b1001c85cc8c95588d66ea63f3
E: ID_SERIAL_SHORT=600508b1001c85cc8c95588d66ea63f3
E: ID_SCSI_SERIAL=PDNMF0ARH1619G
E: multipath=off
E: ID_FS_UUID=8efc2e53-1d6e-4312-a19d-45b89faa6058
E: ID_FS_UUID_ENC=8efc2e53-1d6e-4312-a19d-45b89faa6058
E: ID_FS_VERSION=1.0
E: ID_FS_TYPE=ext4
E: ID_FS_USAGE=filesystem
E: COMPAT_SYMLINK_GENERATION=2
E: DEVLINKS=/dev/disk/by-id/scsi-SHP_LOGICAL_VOLUME_PDNMF0ARH1619G /dev/disk/by-path/pci-0000:0d:00.0-scsi-0:1:0:0 /dev/disk/by-id/wwn-0x600508b1001c85cc8c95588d66ea63f3 /dev/disk/by-id/scsi-3600508b1001c85cc8c95588d66ea63f3 /dev/disk/by-uuid/8efc2e53-1d6e-4312-a19d-45b89faa6058 /dev/disk/by-id/scsi-0HP_LOGICAL_VOLUME_00000000
E: TAGS=:systemd:
````

**lshw -c storage**
```bash
  *-raid                    
       description: RAID bus controller
       product: Smart Array Gen9 Controllers
       vendor: Hewlett-Packard Company
       physical id: 0
       bus info: pci@0000:0d:00.0
       logical name: scsi0
       version: 01
       width: 64 bits
       clock: 33MHz
       capabilities: raid pm msi msix pciexpress bus_master cap_list rom
       configuration: driver=hpsa latency=0
       resources: irq:16 memory:93200000-932fffff memory:93300000-933003ff ioport:7000(size=256) memory:93380000-933fffff
  *-sata
       description: SATA controller
       product: C610/X99 series chipset 6-Port SATA Controller [AHCI mode]
       vendor: Intel Corporation
       physical id: 1f.2
       bus info: pci@0000:00:1f.2
       logical name: scsi5
       version: 05
       width: 32 bits
       clock: 66MHz
       capabilities: sata msi pm ahci_1.0 bus_master cap_list emulated
       configuration: driver=ahci latency=0
       resources: irq:99 ioport:8058(size=8) ioport:807c(size=4) ioport:8050(size=8) ioport:8078(size=4) ioport:8020(size=32) memory:93400000-934007ff
  *-scsi:0
       physical id: 8a
       logical name: scsi10
  *-scsi:1
       physical id: 8b
       logical name: scsi11
  *-scsi:2
       physical id: 8c
       logical name: scsi7
  *-scsi:3
       physical id: 8d
       logical name: scsi8
  *-scsi:4
       physical id: 8e
       logical name: scsi9
```

**lshw -c disk**:
```bash
  *-disk:0                  
       description: SCSI Disk
       product: LOGICAL VOLUME
       vendor: HP
       physical id: 1.0.1
       bus info: scsi@0:1.0.1
       logical name: /dev/sda
       version: 7.00
       serial: PDNMF0ARH1619G
       size: 447GiB (480GB)
       capabilities: 15000rpm gpt-1.00 partitioned partitioned:gpt
       configuration: ansiversion=5 guid=d2a27388-955b-4279-94a1-5a5cb4b256e6 logicalsectorsize=512 sectorsize=512
  *-cdrom
       description: DVD reader
       product: DVDROM DU8DESH
       vendor: HPE
       physical id: 0.0.0
       bus info: scsi@5:0.0.0
       logical name: /dev/cdrom
       logical name: /dev/dvd
       logical name: /dev/sr0
       version: UHJ1
       capabilities: removable audio dvd
       configuration: ansiversion=5 status=nodisc
```

**Comment by irishgordo** [comment]
@ibrokethecloud thanks for the fix :smile: :+1:  - I'm not seeing this on v1.4.0-rc2, so I will go ahead and close this out :smile: 


https://github.com/user-attachments/assets/272eed46-afb4-4147-bc20-ad99a6f6d7f5
